Random offset based on difficulty

FL3SH opened the pull request.

lpanebr commented:
I think this should enable the #72 but I just tried this pr and I's not working for me.

I tried setting `let g:vim_be_good_delete_me_fixed_offset = 1` and `let g:vim_be_good_delete_me_fixed_offset = '1'` but the offset keeps changing..

In the debug I get zero as if the config is not getting read:
```
{"message":"[vim-be-good] delete-round#getGameOptions - isDefinedUserRandomOffset  0","level":"info"}
{"message":"[vim-be-good] delete-round#getGameOptions - isDefinedUserFixedOffset  0","level":"info"}
```

FL3SH commented:
This is why is just a draft. It has a bug and I don't see any issue causing it.
Options are not being read properly, but if you set both random and fixed offsets, fixed values should be picked.

lpanebr commented:
yes, you're right.

I could not find anything wrong with the logic so I thought maybe the logging is somehow messing things and what you know. When I comment it out:
```typescript
        /* console.log( */
        /*     "delete-round#getGameOptions - isDefinedUserRandomOffset ", */
        /*     randomOffset, */
        /* ); */
        /* console.log( */
        /*     "delete-round#getGameOptions - isDefinedUserFixedOffset ", */
        /*     fixedOffset, */
        /* ); */
```
 **the fixed value gets correcty picked from my init.vim!**
```ini
let g:vim_be_good_delete_me_fixed_offset = 3
" let g:vim_be_good_delete_me_random_offset = 1
```

I then tried uncommenting and moving the `console.log` entries before the return but the build failed with:
```bash
Checking formatting...
src/game/delete-round.ts
Code style issues found in the above file(s). Forgot to run Prettier?
```

Since I now nothing about typescript or prettier I can't help anymore.. and plus I've got to get some real work done..

FL3SH commented:
Run `npm run lint:fix` and then build the app.

ThePrimeagen commented:
@FL3SH please let me know when you want me to merge / review this :)

FL3SH commented:
Hey, @ThePrimeagen if you find a moment to look at this code. I sure I am missing sth because vim setting is not being picked up, but I can't find any issues.

Of course, I will try to solve it on my own, but I don't know when I find some time for it, maybe on the weekend.

brandoncc commented on README.md:
```diff
@@ -29,10 +29,18 @@ current buffer (only if its empty) set `vim_be_good_floating` to 0.
 ### Games - relative
 By default vim be good returns random offset for game difficult above noob, if 
-you with to set fixed offset set `vim_be_good_delete_me_offset` to desired
+you with to set fixed offset set `vim_be_good_delete_me_fixed_offset` to desired
```
"you with to set" typo?

brandoncc commented on README.md:
```diff
@@ -29,10 +29,18 @@ current buffer (only if its empty) set `vim_be_good_floating` to 0.
+You can also use `vim_be_good_delete_me_random_offset` to set minimal value of 
+random offset.
+
+`let g:vim_be_good_delete_me_fixed_offset = 5`
```
I think this should say `g:vim_be_good_delete_me_fixed_offset`?

brandoncc commented on src/game/delete-round.ts:
```diff
@@ -14,25 +14,96 @@ export class DeleteRound extends Round {
+        let fixedOffset: number = 0;
+
+        const isDefinedUserRandomOffset = await game.nvim.eval(
+            'exists("vim_be_good_delete_me_fixed_offset")',
```
I debugged this for like an hour, completely confused... Then I realized that lines 24 and 27 are backward!

🤦

FL3SH replied:
omg, sorry @brandoncc ;)

lpanebr replied:
can't believe I also didn't see that! ahhahaha

FL3SH commented:
@lpanebr should be fine now, you can check it ;)

lpanebr commented:
ALL seem VimBeGood now! Excellent work!!!

FL3SH commented on src/game/delete-round.ts:
```diff
@@ -14,25 +14,96 @@ export class DeleteRound extends Round {
+        let fixedOffset = userOptions["vim_be_good_delete_me_fixed_offset"];
+        let offset: number;
+
+        let maxOffset = {
```
is it good enough to keep max/min offset in two objects or should I merge them?

FL3SH replied:
nvm, I feel like I need to change it

brandoncc commented on README.md:
```diff
@@ -120,3 +128,14 @@ idea of how to stop sucking
 - Check your new feature `NVIM_NODE_LOG_FILE=/tmp/nvim.log nvim +VimBeGood` (you can view logs with `less /tmp/nvim.log`)
 - Make PR
+
+If you see something similar during your build:
```
Nice!
